File: Filmes-API/Filmes.DOMAIN/Entity/All/Filme.cs
using Filmes.DOMAIN.Interface;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Filmes.DOMAIN.Entity.All
{
    [Table("Filme", Schema = "All")]
    public class Filme : IEntity
    {

        public long Id { get; set; }
        [MaxLength(45),Required] public string Titulo { get; set; }
        [DataType(DataType.Date)]
        public DateTime? DataLancamento { get; set; }
        [MaxLength(255), Required] public string Sinopse { get; set; }
        public int Avaliacoes { get; set; }
        public string Genero { get; set; }
        public byte[] Poster { get; set; }
        public ICollection<FilmeDiretor> Diretores { get; set; }
        public Filme()
        {
            Diretores = new List<FilmeDiretor>();
        }
    }
}

File: Filmes-FRONTEND/src/core/models/all/Filme.ts
import { Diretor } from ".";

export class Filme implements Shared.IEntity{
    id: number = 0;
    titulo: string = '';
    dataLancamento: string = '';
    sinopse: string = '';
    avaliacoes: number = 0;
    genero: string = '';
    poster: any = null;
    diretores: Filme[]=[];

    constructor(model?: Filme){

        if(!model)
            return;

        this.id = model.id;
        this.titulo = model.titulo;
        this.dataLancamento = model.dataLancamento;
        this.sinopse = model.sinopse;
        this.avaliacoes = model.avaliacoes;
        this.poster = model.poster;
        this.diretores = model.diretores;
        
    }
}